pkg/network/usm_connection_keys.go

```diff
@@ -75,13 +75,13 @@ func WithKey(connectionStats ConnectionStats, f func(key types.ConnectionKey) (s
 		clientIPNAT, clientPortNAT, serverIPNAT, serverPortNAT = serverIPNAT, serverPortNAT, clientIPNAT, clientPortNAT
 	}
 
-	// Callback 1: NATed (client, server)
-	if hasNAT && f(types.NewConnectionKey(clientIPNAT, serverIPNAT, clientPortNAT, serverPortNAT)) {
+	// Callback 1: (client, server)
+	if f(types.NewConnectionKey(clientIP, serverIP, clientPort, serverPort)) {
```

Comment:
I suggest adding a comment here explaining the reason for the change, to help prevent future breaking changes